src/XliffTasks/StringExtensions.cs

```diff
@@ -42,9 +43,9 @@ public static int GetReplacementCount(this string text)
         {
             int replacementCount = 0;
 
-            foreach (Match placeholder in s_placeholderRegex.Matches(text))
+            foreach (Match placeholder in s_placeholderRegex.Matches(text).Cast<Match>())
```

Comment:
Why is the `.Cast<>()` needed? What type was being returned instead?

Reply:
Actually, that's interesting now that you point it out. I was just blindly following the analyzer -- lemme dig into it.

Reply:
Apparently it returns `object`

Reply:
ah, I see. `.Matches` returns a `MatchCollection`, whose members are implicitly cast by the `foreach` into `Match`s but are apparently `object`s under the covers. Interesting.

Reply:
😕

Reply:
I can remove the cast if it makes you sad haha

Reply:
Nah I trust ya